src/main/java/racingcar/util/RetryableTemplate.java

```diff
@@ -0,0 +1,16 @@
+package racingcar.util;
+
+import java.util.function.Consumer;
+
+public class RetryableTemplate {
+
+    public void execute(Runnable runnable, Consumer<Exception> exceptionHandler) {
+        try {
+            runnable.run();
+            return;
+        } catch (Exception e) {
+            exceptionHandler.accept(e);
+        }
+        execute(runnable, exceptionHandler);
+    }
```

Comment:
해당 코드는 적절한 위치에 적절한 코드를 위치시킨다면 더 가독성 좋고 간결하게 수정이 가능할 것 같아요!! 이 부분은 제가 직접 알려드려도 괜찮을 것 같아서 아래 제안하는 코드를 남겨두도록 할게요!!

```suggestion
    public void execute(Runnable runnable, Consumer<Exception> exceptionHandler) {
        try {
            runnable.run();
        } catch (Exception e) {
            exceptionHandler.accept(e);
            execute(runnable, exceptionHandler);
        }
    }
```

Reply:
추가적으로 해당 코드는 예외 발생 상황(입력의 오류)일 경우 다시금 본인을 호출하는 재귀 구조로 구현되어 있습니다.

만약 누군가가 악의를 품고 수백, 수천만번의 예외를 발생시킨다면 어떻게 될까요?? 함수 호출 스택은 어디까지 쌓일까요?? 이런 부분에 대해서 고민해보고 한번 개선할 수 있는 방향으로 코드를 작성해보는 것도 많은 도움이 될 것 같아요!!!

Reply:
스티치가 말씀하신 상황이 발생한다면 StackOverFlow 같은 문제가 발생하는 걸까요? 그럼 결국 재귀를 사용하지 않아야 하는 걸까요?

Reply:
재귀를 사용한다면 코드가 간결하고 명시적이라는 장점이 있지만 스택이 계속 쌓여서 메모리 초과가 발생할 수 있다는 단점도 있습니다!!
혹시 알고리즘을 해보셨으면 아실 수도 있지만 재귀는 반복을 통해서 해결을 할 수도 있을 거에요!!

한 번 해당 부분을 찾아보고 고민해보시면 좋을 것 같아요!!

추가적으로 혹시나 오해하실 수도 있을 것 같아서 첨언하지면 '재귀를 사용한 코드가 좋지 않다'라는 이야기는 아닙니다!! 저도 알고리즘을 할 때 재귀를 사용하는 것을 매우 좋아했거든요!! 하지만 프로그램을 만드는 경우 가독성이 너무 떨어지는 문제를 만드는 것이 아니라면 최대한 안정적인 프로그램을 만드는 것이 더 중요하다고 생각해요. 그런 관점에서 해당 부분에 대한 고민을 해 보시길 추천드렸습니다!! 

이 또한 저의 관점이고 저의 생각이기 때문에 조조그린이 이러한 저의 생각을 참고하면서 본인의 코드에 대한 관점을 만들어 나가면 좋을 것 같아요!!